src/lib/interfaces/profile-upload-options.ts

```diff
@@ -0,0 +1,15 @@
+import { Options } from 'ipfs-http-client';
+
+export interface ProfileUploadOptionsHTTP {
+  url: string;
+  port: number;
+  ipfsClientOptions?: never;
+}
+
+export interface ProfileUploadOptionsIPFS {
+  url?: never;
+  port?: never;
+  ipfsClientOptions: Options;
+}
+
```

Comment:
I would suggest to use a discriminated union type and extend from a base: https://betterprogramming.pub/advanced-typescript-how-to-use-interface-inheritance-with-discriminated-unions-dddf77cb3836

Something like this:

```typescript
enum UPLOAD_TYPE {
    HTTP,
    IPFS
}

interface ProfileUploadOptionsBase {
    type: UPLOAD_TYPE,
    url?: string;
    port?: number;
    ipfsClientOptions?: Options;
}

export interface ProfileUploadOptionsHTTP extends ProfileUploadOptionsBase {
    type: UPLOAD_TYPE.HTTP
    url: string;
    port: number;
    ipfsClientOptions?: never;
  }
  
  export interface ProfileUploadOptionsIPFS extends ProfileUploadOptionsBase {
    type: UPLOAD_TYPE.IPFS,
    url?: never;
    port?: never;
    ipfsClientOptions: Options;
  }
```

It might be a bit too much but I think this use case is very suitable for a discriminated union types.

Reply:
There isn't really any overlap to justify creating a base imo, this is just duplicating code. Also this would require the person using the lib to provide the type property, which is redundant due to separating the ipfs options into a separate property `ipfsClientOptions`.

What would be the benefit of doing it this way? Are you thinking ahead if we might get another "upload type"?

I think it's good to think about this, but we can tackle this with the next PR.

Reply:
ok yes indeed you are right ;)